src/bentoml_cli/deployment.py

```diff
@@ -127,23 +155,35 @@ def deployment_cli():
     @click.option(
         "-f",
         "--file",
-        type=click.Path(exists=True),
+        type=click.File(),
         default=None,
-        help="Create deployment using json file",
+        help="Create deployment using json file. ",
         cls=Mutex,
-        not_required_if=["deployment_name"],
+        must_exist_if_none=["deployment_name", 'bento'],
+        cannot_exist_together=[
+            "deployment_name",
+            "bento",
+            "description",
+            "expose_endpoint",
+            "kube_namespace",
+            "resource_instance",
+            "min_replicas",
+            "max_replicas",
+            "type",
+            "mode",
+            "runner",
+            "force",
+            "threads",
+            "push"
+        ],
```

Comment:
I still think having to implement this as a eager feature works the best. This is not ideal imho

Reply:
The issue with eager is that it is unable to check if `--file` and `--deployment_name` are provided together (we only want one to be provided). I will check if we can add that check somehow with eager.

Reply:
if --file is passed then it ignore everything else if it is an eager option right? we confirmed that behaviour the other day?

Reply:
Yes that's right, but eager is processed first so it cannot disallow commands such as `bentoml deployment create -f create.json --deployment-name test-1`. However `--deployment-name` here is omitted. Not sure if we want to forbid extra commands when `-f/--file` is present? Currently it'll give a warning of `Error: Illegal usage: 'file' is mutually exclusive with 'deployment_name'` in that situation.

Reply:
What I'm saying is in the case of `--file` user need to put into the environment variable for deployment-name and cluster-name

Reply:
```bash
BENTOCLOUD_DEPLOYMENT_NAME=... BENTOCLOUD_CLUSTER_NAME=... bentoml deploy -f create.json
```